spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -119,10 +122,16 @@ public ExecutorProvider subscriberExecutorProvider() {
 	}
 
 	@Bean
+	@ConditionalOnBean(type = "com.fasterxml.jackson.databind.ObjectMapper")
 	@ConditionalOnMissingBean
-	public PubSubMessageConverter pubSubMessageConverter() {
-		return new JacksonPubSubMessageConverter(
-				this.gcpPubSubProperties.getTrustedPackages());
+	public JacksonPubSubMessageConverter pubSubMessageConverter(ObjectMapper objectMapper) {
+		return new JacksonPubSubMessageConverter(objectMapper);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SimplePubSubMessageConverter pubSubMessageConverter() {
```

Comment:
How does it work if you change both return types to the interface  - `PubSubMessageConverter` ?

Reply:
I thought I tested that before, but now it seems to work.